src/shared/apis/queryCache.ts

```diff
@@ -229,18 +220,14 @@ export const getCachedResultsOrRunQuery = (
       cancel: () => {},
     }
   }
-  const variableAssignments = variables
-    .map(v => asAssignment(v))
-    .filter(v => !!v)
 
-  let windowVars = []
-
-  if (hasWindowVars(variableAssignments) === false) {
-    windowVars = getWindowVarsFromVariables(query, variables)
-  }
+  const windowVar = getWindowPeriodVariableFromVariables(query, variables)
 
   // otherwise query & set results
-  const extern = buildUsedVarsOption(query, variables, windowVars)
+  const extern = buildUsedVarsOption(
+    query,
+    windowVar ? variables.concat(windowVar) : variables
+  )
```

Comment:
`buildUsedVarsOption` 
* already has a contract to de-dupe based upon what variables are in the query.
* it then appends the optional 3rd function parameter.

Reply:
note: the `buildUsedVarsOption` dedupe is done using the older (treat everything as global scope) pattern. But we can fix that is a followup/future PR.